python/google/protobuf/internal/well_known_types.py

```diff
@@ -372,6 +399,34 @@ def _NormalizeDuration(self, seconds, nanos):
     self.seconds = seconds
     self.nanos = nanos
 
+  def _comparable_scalar(self, obj):
+    if isinstance(obj, timedelta):
+      return (obj.seconds + obj.days * _SECONDS_PER_DAY) * _NANOS_PER_SECOND + obj.microseconds * _NANOS_PER_MICROSECOND
```

Comment:
An issue with both this, and the comparison for datetimes (along with switching between types) is that their resolution is different (builtin milliseconds vs. protobuf nanoseconds). This is probably something to make sure that is documented, mostly for the comparisons.